src/System.Device.Gpio/System/Device/Gpio/Drivers/Libgpiod/LibGpiodDriverFactory.cs

```diff
@@ -0,0 +1,161 @@
+            }
+            else if (_driverVersionEnvVarValue == LibGpiodDriverVersion.V2.ToString())
+            {
+                _driverVersionSetByEnvVar = LibGpiodDriverVersion.V2;
+            }
+        }
+    }
+
+    public static VersionedLibgpiodDriver Create(int chipNumber)
+    {
+        if (_driverVersionEnvVarValue != null)
+        {
+            if (_driverVersionSetByEnvVar == null)
+            {
+                throw new GpiodException($"Can not create libgpiod driver due to invalid specified value in environment variable" +
+                    $" {DriverVersionEnvVar}: '{_driverVersionEnvVarValue}'" +
+                    $". Valid values: {string.Join(", ", _libraryToDriverVersionMap.Values.Distinct())}");
+            }
+
+            var version = _driverVersionSetByEnvVar.Value;
+            var driver = CreateInternal(_driverVersionSetByEnvVar.Value, chipNumber);
+
+            return new VersionedLibgpiodDriver(version, driver);
+        }
+
+        return CreateAutomaticallyChosenDriver(chipNumber);
+    }
+
+    public static VersionedLibgpiodDriver Create(int chipNumber, LibGpiodDriverVersion driverVersion)
+    {
+        return new VersionedLibgpiodDriver(driverVersion, CreateInternal(driverVersion, chipNumber));
+    }
+
+    private static VersionedLibgpiodDriver CreateAutomaticallyChosenDriver(int chipNumber)
+    {
+        if (_automaticallySelectedDriverVersion == null)
+        {
+            throw new GpiodException($"No supported libgpiod library file found.\n" +
+                $"Supported versions: {string.Join(", ", _libraryToDriverVersionMap.Keys)}\n" +
+                $"Searched paths: {string.Join(", ", _librarySearchPaths)}");
+        }
+
+        var version = _automaticallySelectedDriverVersion.Value;
+        var driver = CreateInternal(_automaticallySelectedDriverVersion.Value, chipNumber);
+
+        return new VersionedLibgpiodDriver(version, driver);
+    }
+
+    private static GpioDriver CreateInternal(LibGpiodDriverVersion version, int chipNumber)
+    {
+        if (!_driverCandidates.Contains(version))
+        {
+            throw new GpiodException($"No suitable libgpiod library found for {nameof(LibGpiodDriverVersion)}.{version}. " +
+                $"Supported versions: {string.Join(", ", _driverVersionToLibrariesMap[version])}\n" +
+                $"Searched paths: {string.Join(", ", _librarySearchPaths)}");
+        }
+
+        return version switch
+        {
+            LibGpiodDriverVersion.V1 => new LibGpiodV1Driver(chipNumber),
+            LibGpiodDriverVersion.V2 => new LibGpiodV2Driver(LibGpiodProxyFactory.CreateChip(chipNumber)),
+            _ => throw new ArgumentOutOfRangeException(nameof(version), version, null)
+        };
+    }
+
+    private static IEnumerable<string> GetInstalledLibraries()
+    {
+        HashSet<string> foundLibrariesFileName = new();
+
+        foreach (string searchPath in _librarySearchPaths)
```

Comment:
This hardcoded search path list is a bit ugly. Can we use $path instead?

Reply:
$PATH does not point to libraries (usually) but to directories containing executables.
In [this comment](https://github.com/dotnet/iot/pull/2184#discussion_r1418766214) we discussed what to do about this uglyness but have not come up with a better approach (yet)